addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/handler/SenseBoxHandler.java

```diff
@@ -0,0 +1,372 @@
+            }
+        } else {
+            logger.debug("Cannot handle command. Thing is not ONLINE.");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        refreshJob.cancel(true);
+    }
+
+    private void startAutomaticRefresh() {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                logger.info("Refreshing data for box {}, scheduled after {} seconds...",
+                        thingConfiguration.getSenseBoxId(), thingConfiguration.getRefreshInterval());
+
+                try {
+                    boolean success = fetchData();
+                    if (success) {
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_NAME));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_LOCATION));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_EXPOSURE));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_API_URL));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_MAP_URL));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_IMAGE_URL));
+
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_UV_INTENSITY));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_LUMINANCE));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_PRESSURE));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_HUMIDITY));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_TEMPERATURE));
+
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_UV_INTENSITY_LR));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_LUMINANCE_LR));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_PRESSURE_LR));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_HUMIDITY_LR));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_TEMPERATURE_LR));
+                    }
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            }
+        };
+
+        refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, thingConfiguration.getRefreshInterval(),
+                TimeUnit.SECONDS);
+    }
+
+    private boolean fetchData() {
+        try {
+            data = CACHE.get(thingConfiguration.getSenseBoxId());
+        } catch (IOException e) {
+            logger.error("Exception occurred during fetching data: {}", e.getMessage(), e);
```

Comment:
Aaaaah.... Oops 😊 

Yes, it might be that I might have forgotten to run that 'git push --force' 😄 

Now I did it, thank you for your patience with this PR, I hope we whipped it into mergeable state now.